internal/feather.go

```diff
@@ -91,7 +91,19 @@ func convertReflectToArrowType(value reflect.Type) (arrow.DataType, error) {
 	}
 }
 
-func setBuilderValues(builder array.Builder, slice reflect.Value, nullMask []bool) error {
+func setBuilderValues(builder array.Builder, slice reflect.Value, valid []bool) error {
+	if valid == nil {
```

Comment:
could we require the caller to always initialize the valid array, or return this from the function? it's a bit confusion how it will be created if null

Reply:
good point - I will make the caller pass it in